core/arch/disassemble_shared.c

```diff
@@ -887,13 +894,19 @@ instr_disassemble_opnds_noimplicit(char *buf, size_t bufsz, size_t *sofar INOUT,
                          */
                         optype = 0;
                     });
-        bool is_mask = !dsts_first() && !instr_is_opmask(instr) && opnd_is_reg(opnd) &&
-            reg_is_opmask(opnd_get_reg(opnd));
-        print_to_buffer(buf, bufsz, sofar, is_mask ? "{" : "");
-        printing =
-            opnd_disassemble_noimplicit(buf, bufsz, sofar, dcontext, instr, optype, opnd,
-                                        prev, multiple_encodings, dsts_first(), &i);
-        print_to_buffer(buf, bufsz, sofar, is_mask ? "}" : "");
+        bool is_evex_mask = !instr_is_opmask(instr) && opnd_is_reg(opnd) &&
+            reg_is_opmask(opnd_get_reg(opnd)) && opmask_with_dsts();
+        if (!is_evex_mask) {
+            print_to_buffer(buf, bufsz, sofar, is_evex_mask ? " {" : "");
+            printing = opnd_disassemble_noimplicit(buf, bufsz, sofar, dcontext, instr,
+                                                   optype, opnd, prev && !is_evex_mask,
+                                                   multiple_encodings, dsts_first(), &i);
+            print_to_buffer(buf, bufsz, sofar, is_evex_mask ? "}" : "");
+        } else if (is_evex_mask) {
+            CLIENT_ASSERT(!dsts_first(), "evex mask can only be a source.");
+            CLIENT_ASSERT(!is_evex_mask_pending, "There can only be one evex mask.");
```

Comment:
nit: make these 2 lines consistent: one capitalizes the first word while the other does not

Reply:
Ok (did that because I didn't want to spell Evex - I thought it looks strange). Done